test/instrumentation.ts

```diff
@@ -0,0 +1,184 @@
+/*!
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as assert from 'assert';
+import {Entry} from '../src';
+import * as instrumentation from '../src/utils/instrumentation';
+import {google} from '../protos/protos';
+
+const NAME = 'name';
+const VERSION = 'version';
+const NODEJS_TEST = instrumentation.NODEJS_LIBRARY_NAME_PREFIX + '-test';
+const LONG_NODEJS_TEST =
+  instrumentation.NODEJS_LIBRARY_NAME_PREFIX + '-test-ooooooooooooooo';
+const VERSION_TEST = '1.0.0';
+const LONG_VERSION_TEST = VERSION_TEST + '.0.0.0.0.0.0.0.0.11.1.1-ALPHA';
+
+describe('instrumentation_info', () => {
+  beforeEach(() => {
+    instrumentation.resetInstrumentationStatus();
+  });
+
+  it('should generate library info properly by default', () => {
+    console.log('The version is: ' + instrumentation.getNodejsLibraryVersion());
+    const entry = instrumentation.createDiagnosticEntry(
+      undefined,
+      undefined
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    ) as any;
+    assert.equal(
+      entry.data?.[instrumentation.DIAGNOSTIC_INFO_KEY]?.[
+        instrumentation.INSTRUMENTATION_SOURCE_KEY
+      ]?.[0]?.[NAME],
+      instrumentation.NODEJS_LIBRARY_NAME_PREFIX
+    );
+    assert.equal(
+      entry.data?.[instrumentation.DIAGNOSTIC_INFO_KEY]?.[
+        instrumentation.INSTRUMENTATION_SOURCE_KEY
+      ]?.[0]?.[VERSION],
+      instrumentation.getNodejsLibraryVersion()
+    );
+  });
+
+  it('should add instrumentation log entry to the list', () => {
+    const dummyEntry = createEntry(undefined, undefined);
+    const entries = instrumentation.populateInstrumentationInfo(dummyEntry);
+    assert.equal(entries.length, 2);
+    assert.deepEqual(dummyEntry, entries[0]);
+    assert.equal(
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      (entries[1] as any).data?.[instrumentation.DIAGNOSTIC_INFO_KEY]?.[
+        instrumentation.INSTRUMENTATION_SOURCE_KEY
+      ]?.[0]?.[NAME],
+      instrumentation.NODEJS_LIBRARY_NAME_PREFIX
+    );
+  });
+
+  it('should add instrumentation info to existing list', () => {
+    const dummyEntry = createEntry(NODEJS_TEST, VERSION_TEST);
+    const entries = instrumentation.populateInstrumentationInfo(dummyEntry);
+    assert.equal(entries.length, 1);
+    assert.equal(
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
To get rid of this linter, maybe this help:
```
entries.forEach(entry => {
 assert(entry instanceof Entry);
 // do the rest of equal check;
```

Reply:
Seems that I had that conversion as leftover from previously written code - removed casting completely, thanks!